src/main/java/hudson/plugins/ec2/EC2Computer.java

```diff
@@ -92,19 +94,53 @@ public SlaveTemplate getSlaveTemplate() {
      * Gets the EC2 console output.
      */
     public String getConsoleOutput() throws AmazonClientException {
-        AmazonEC2 ec2 = getCloud().connect();
-        GetConsoleOutputRequest request = new GetConsoleOutputRequest(getInstanceId());
-        return ec2.getConsoleOutput(request).getOutput();
+        try {
+            return getDecodedConsoleOutputResponse().getOutput();
+        } catch (InterruptedException e) {
+            return null;
+        }
     }
 
     /**
      * Gets the EC2 decoded console output.
      * @since TODO
      */
     public String getDecodedConsoleOutput() throws AmazonClientException {
+        try {
+            return getDecodedConsoleOutputResponse().getDecodedOutput();
+        } catch (InterruptedException e) {
+            return null;
+        }
+    }
+
+    private GetConsoleOutputResult getDecodedConsoleOutputResponse() throws AmazonClientException, InterruptedException {
         AmazonEC2 ec2 = getCloud().connect();
         GetConsoleOutputRequest request = new GetConsoleOutputRequest(getInstanceId());
-        return ec2.getConsoleOutput(request).getDecodedOutput();
+        if (checkIfNitro()) {
+            //Can only be used if instance has hypervisor Nitro
+            request.setLatest(true);
+        }
+        return ec2.getConsoleOutput(request);
+    }
+
+    /**
+     * Check if instance has hypervisor Nitro
+     */
+    private boolean checkIfNitro() throws AmazonClientException, InterruptedException {
+        if (isNitro == null) {
+            DescribeInstanceTypesRequest request = new DescribeInstanceTypesRequest();
```

Comment:
Unfortunately the only way I could figure out how do see if an instance is nitro based is via describe-instance-types, describe-instance dont show it. Hypervisor is there but it will not show nitro.

Reply:
alright should we make it more backwards compatible by wrapping `checkIfNitro` with a try catch such that if the underlying request fails it returns false? in case someone does not have the `DescribeInstanceTypes` permission

Reply:
Good point. Added a try catch AmazonClientException which then will default to false if IAM permissions is lacking.